scripts/conversion_script.py

```diff
@@ -0,0 +1,477 @@
+
+    def __init__(self, message, report):
+        super(ProcessError, self).__init__(report)
+        self.message = message
+        self.report = report
+
+
+class OutputCollector(object):
+    """Wrapper class for script expected stdout"""
+
+    def __init__(self, status="", message="", report="", entries=None):
+        self.status = status
+        self.message = message
+        self.report = report
+        self.tasks_format_version = "1.0"
+        self.tasks_format_id = "oamg-format"
+        self.entries = entries
+        self.report_json = None
+
+    def to_dict(self):
+        # If we have entries, then we change report_json to be a dictionary
+        # with the needed values, otherwise, we leave it as `None` to be
+        # transformed to `null` in json.
+        if self.entries:
+            self.report_json = {
+                "tasks_format_version": self.tasks_format_version,
+                "tasks_format_id": self.tasks_format_id,
+                "entries": self.entries,
+            }
+
+        return {
+            "status": self.status,
+            "message": self.message,
+            "report": self.report,
+            "report_json": self.report_json,
+        }
+
+
+def find_highest_report_level(actions):
+    """
+    Gather status codes from messages and result. We are not seeking for
+    differences between them as we want all the results, no matter from where
+    they come.
+    """
+    print("Collecting and combining report status.")
+    action_level_combined = []
+    for value in actions.values():
+        action_level_combined.append(value["result"]["level"])
+        for message in value["messages"]:
+            action_level_combined.append(message["level"])
+
+    valid_action_levels = [
+        level for level in action_level_combined if level in STATUS_CODE
+    ]
+    valid_action_levels.sort(key=lambda status: STATUS_CODE[status], reverse=True)
+    return valid_action_levels[0]
+
+
+def gather_json_report():
+    """Collect the json report generated by convert2rhel."""
+    print("Collecting JSON report.")
+    with open(C2R_REPORT_FILE, "r") as handler:
+        data = json.load(handler)
+
+    if not data:
+        raise ProcessError(
+            message="Expected content missing in the pre-conversion analysis report.",
+            report="The convert2rhel analysis report file '%s' does not contain any JSON data in it."
+            % C2R_REPORT_FILE,
+        )
+
+    return data
+
+
+def gather_textual_report():
+    """Collect the textual report generated by convert2rhel."""
+    data = ""
+    if os.path.exists(C2R_REPORT_TXT_FILE):
+        with open(C2R_REPORT_TXT_FILE, mode="r") as handler:
+            data = handler.read()
+    return data
+
+
+def generate_report_message(migration_result, highest_status):
+    """Generate a report message based on the status severity."""
+    message = ""
+    if STATUS_CODE[highest_status] <= STATUS_CODE["WARNING"] and migration_result:
```

Comment:
This line confuses me little bit, I can have pre-conversion report (not conversion report?), we will find the highest status and let's say it will be lower than warning, but if the migration_results is False (e.g. the file doesn't exist), then this will be false, and whole message ends up empty? Maybe it's corner case that wouldn't happen?

Reply:
So, thinking more about it, I don't think that `migration_result` is entirely necessary here... I guess that we can get away with checking the status in the report. It is redundant information, after all (Also, simplifies the code).

Reply:
sure, that also seems acceptable and simple!